macros/edr/tests/test_collect_metrics.sql

```diff
@@ -35,12 +35,16 @@
     {% set table_metrics = [] %}
     {% set col_to_metrics = {} %}
     {% for metric in metrics %}
+        {% if not metric.name %}
```

Comment:
Should we ensure the metric's name is unique across the entire project, or allow the same metric name to be calculated from different tests?

Reply:
Maybe we should check that it's unique in the **model**?
Since in any way in the cloud the test is defined on a specific model (or source) - so I don't really care if the same metric name appears across models.
But I do want to avoid relying on the dbt test ID for filtering the metric (since it's sensitive to the test parameters)

Reply:
Added [here](https://github.com/elementary-data/dbt-data-reliability/pull/736/commits/ce97140a6f6487dc7a4dd39e3d522d124b282c78).

I originally implemented it using the `cache` and therefore the changes,
but it made sense for me that they'd live in the same module so I kept it.